internal/pkg/cli/job_init.go

```diff
@@ -323,6 +323,25 @@ func (o *initJobOpts) askSchedule() error {
 	return nil
 }
 
+func (o initJobOpts) legitimizePlatform() error {
+	detectedOs, detectedArch, err := o.dockerEngine.GetPlatform()
+	if err != nil {
+		return fmt.Errorf("get docker engine platform: %w", err)
+	}
+	detectedPlatform := aws.String(manifest.PlatformString(detectedOs, detectedArch))
+	platform, err := manifest.RedirectPlatform(o.image, detectedOs, detectedArch, o.wkldType)
```

Comment:
It's a bit odd that `RedirectPlatform` takes an image as input to only shortcircuit. Can we remove `image` as a param from `RedirectPlatform` and add it as the first if-check in `legitimizePlatform` to shortcircuit?